File: app/src/main/java/com/wang/example/msg/bean/TestData.java
package com.wang.example.msg.bean;

import com.alibaba.fastjson.JSON;
import com.wang.example.R;

import java.util.ArrayList;
import java.util.List;

public class TestData {

    /**
     * 生成测试数据
     */
    public static List<BaseMsgBean> createMsgList() {
        ArrayList<BaseMsgBean> list = new ArrayList<>();
        for (int i = 0; i < 20; i++) {
            String json;
            switch (i % 5) {
                case 0:
                    json = createText(i);
                    break;
                case 1:
                    json = createImg(i);
                    break;
                case 2:
                    json = createOrder(i, 1);
                    break;
                case 3:
                    json = createOrder(i, 2);
                    break;
                default:
                    json = createOrder(i, 3);
                    break;
            }
            list.add(JSON.parseObject(json, BaseMsgBean.class));
        }
        return list;
    }

    private static String createText(int position) {
        return "{\"msgType\":1,\"data\":\"{\\\"text\\\":\\\"" + position + "\\\"}\"}";
    }

    private static String createImg(int position) {
        return "{\"msgType\":2,\"data\":\"{\\\"imgRes\\\":" + R.mipmap.ic_launcher + "}\"}";
    }

    private static String createOrder(int position, int orderType) {
        return "{\"msgType\":3,\"data\":\"{\\\"orderType\\\":" + orderType + ",\\\"orderNo\\\":\\\"0000" + position + "\\\",\\\"orderName\\\":\\\"哈哈\\\",\\\"otherOrderData\\\":{\\\"emsNo\\\":\\\"11\\\"}}\"}";
    }
}

File: app/src/main/java/com/wang/example/msg/bean/TestData.kt
package com.wang.example.msg.bean

import com.google.gson.Gson
import com.wang.example.R
import com.wang.example.msg.bean.BaseMsgBean

object TestData {
    /**
     * 生成测试数据
     */
    @JvmStatic
    fun createMsgList(): List<BaseMsgBean> {
        val list = ArrayList<BaseMsgBean>()
        for (i in 0..19) {
            val json = when (i % 5) {
                0 -> createText(i)
                1 -> createImg(i)
                2 -> createOrder(i, 1)
                3 -> createOrder(i, 2)
                else -> createOrder(i, 3)
            }
            list.add(Gson().fromJson(json, BaseMsgBean::class.java))
        }
        return list
    }

    private fun createText(position: Int): String {
        return "{\"msgType\":1,\"data\":\"{\\\"text\\\":\\\"$position\\\"}\"}"
    }

    private fun createImg(position: Int): String {
        return "{\"msgType\":2,\"data\":\"{\\\"imgRes\\\":" + R.mipmap.ic_launcher + "}\"}"
    }

    private fun createOrder(position: Int, orderType: Int): String {
        return "{\"msgType\":3,\"data\":\"{\\\"orderType\\\":$orderType,\\\"orderNo\\\":\\\"0000$position\\\",\\\"orderName\\\":\\\"哈哈\\\",\\\"otherOrderData\\\":{\\\"emsNo\\\":\\\"11\\\"}}\"}"
    }
}